### PR TITLE
chore: fix doc comments

### PR DESCRIPTION
**Pull request by deekshas8**
This PR fixes more issues in doc comments completeness. This is a follow-up on #2221

### CODE REVIEW

File: packages/odata-common/src/request-builder/delete-request-builder-base.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export abstract class DeleteRequestBuilderBase<
 
   /**
    * Instructs the request to force an overwrite of the entity by sending an 'If-Match: *' header instead of sending the ETag version identifier.
-   * @returns this The request itself to ease chaining while executing the request
+   * @returns This The request itself to ease chaining while executing the request.
```
Comment (jjtang1985) [inline packages/odata-common/src/request-builder/delete-request-builder-base.ts:52]:
```suggestion
   * @returns The request itself to ease chaining while executing the request.
```

File: packages/odata-v4/src/uri-conversion/uri-value-converter.ts [hunk 2]
```diff
@@ -2,6 +2,6 @@ import { createUriConverter } from '@sap-cloud-sdk/odata-common/internal';
 import { defaultDeSerializers } from '../de-serializers';
 
 /**
- * internal
+ * @internal
```
Comment (jjtang1985) [inline packages/odata-v4/src/uri-conversion/uri-value-converter.ts:5]: I guess this is not internal, as I see it's exported in the index.

File: packages/util/src/equal.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { isNullish } from './nullish';
 // eslint-disable-next-line valid-jsdoc
 /**
- * hidden
+ * Hidden.
```
Comment (jjtang1985) [inline packages/util/src/equal.ts:4]:
```suggestion
 * @hidden.
```
See doc: https://typedoc.org/guides/doccomments/#%40hidden-and-%40ignore
Please also adjust other occurrences.

File: packages/odata-v4/src/filter-functions.ts [hunk 4]
```diff
@@ -152,7 +152,7 @@ export function hasSubsequence<
   return filterFunction('hassubsequence', 'boolean', subsequence, sequence);
 }
 /**
- * TODO: totalseconds, time
+ * TODO: totalseconds, time.
```
Comment (jjtang1985) [inline packages/odata-v4/src/filter-functions.ts:155]: General question: Is it a manual effort to add the "."?

Comment (deekshas8) [reply packages/odata-v4/src/filter-functions.ts:155]: No, the jsdoc rule `jsdoc/require-description-complete-sentence` does this automatically. It was added in the other PR #2214 . I fixed it in a separate PR so its easier to review

File: packages/odata-v4/src/uri-conversion/uri-value-converter.ts (hunk 2 above)
Comment (deekshas8) [reply packages/odata-v4/src/uri-conversion/uri-value-converter.ts]: Thanks, I thought it was a typo.

File: packages/util/src/equal.ts (hunk 3 above)
Comment (deekshas8) [reply packages/util/src/equal.ts:4]: We export everything from util package, so I am not sure if we need `@hidden` at all, atleast here. I will remove this and adjust in other places

Comment (jjtang1985) [reply packages/util/src/equal.ts]: This hidden is for hiding the api for doc generation.
We can also remove the "hidden" and add valid docs.

File: packages/odata-v4/src/filter-functions.ts (hunk 4 above)
Comment (jjtang1985) [reply packages/odata-v4/src/filter-functions.ts:155]: I guess you enabled this rule then run the lint fix.
The rule was removed for making the pipeline happy in the previous PR, right?
Do we plan to add the rule in the future when everything is fixed?

**Pull request merged**